app/models/category_user.rb

```diff
@@ -23,20 +19,31 @@ def self.watching_levels
   end
 
   def self.batch_set(user, level, category_ids)
-    records = CategoryUser.where(user: user, notification_level: notification_levels[level])
-    old_ids = records.pluck(:category_id)
-    changed = false
-
-    category_ids = Category.where('id in (?)', category_ids).pluck(:id)
-
-    remove = (old_ids - category_ids)
-    if remove.present?
-      records.where('category_id in (?)', remove).destroy_all
-      changed = true
-    end
-
-    (category_ids - old_ids).each do |id|
-      CategoryUser.create!(user: user, category_id: id, notification_level: notification_levels[level])
+    level_num = notification_levels[level]
+    category_ids = Category.where(id: category_ids).pluck(:id)
+
+    # Update pre-existing category users
+    changed =
+      CategoryUser
+        .where(user_id: user.id, category_id: category_ids)
+        .update_all(notification_level: level_num) > 0
+
+    # Remove extraneous category users
+    changed ||=
+      CategoryUser
+        .where(user_id: user.id, notification_level: level_num)
+        .where.not(category_id: category_ids)
+        .delete_all() > 0
```

Comment:
```suggestion
        .delete_all > 0
```

Reply:
This is marked as resolved yet I'm still seeing `delete_all()` - is this github being weird?

Reply:
Oh, I see what happened. I committed the change using github's new commit suggestion feature, but then I didn't pull before rebasing.